### PR TITLE
qt/6.x.x - Adding cross build support (only tested for Windows so far)

### PR DESCRIPTION
**Pull request by szigetics**
Specify library name and version:  **qt/6.x.x**

qt/6.x.x - Adding cross build support (only tested for Windows so far)

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/qt/6.x.x/conanfile.py [hunk 1]
```diff
@@ -616,6 +616,13 @@ def _configure_cmake(self):
         self._cmake.definitions["FEATURE_pkg_config"] = "ON"
         if self.settings.compiler == "gcc" and self.settings.build_type == "Debug" and not self.options.shared:
             self._cmake.definitions["BUILD_WITH_PCH"]= "OFF" # disabling PCH to save disk space
+            
+        if tools.cross_building(self):
```
Comment (madebr) [inline recipes/qt/6.x.x/conanfile.py:620]: @ericLemanissier 
I'm in no way a qt specialist.
I remember reading to cross build qt for android, you first need a native qt.
Would the following be needed instead? + setting the correct appropriate arguments.
```python
def build_requirements(self):
    if tools.cross_building(self):
        self.build_requires(f"{self.name}/{self.version}")
```

Comment (ericLemanissier) [reply recipes/qt/6.x.x/conanfile.py:620]: that's the way qt 6 handles cross compilation yes, but I don't think conan allows it.

File: recipes/qt/6.x.x/conanfile.py [hunk 2]
```diff
@@ -691,19 +698,11 @@ def package(self):
             if module != "qtbase" and not self.options.get_safe(module):
                 tools.rmdir(os.path.join(self.package_folder, "licenses", module))
         tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
-        for mask in ["Find*.cmake", "*Config.cmake", "*-config.cmake"]:
-            tools.remove_files_by_mask(self.package_folder, mask)
```
Comment (ericLemanissier) [inline recipes/qt/6.x.x/conanfile.py:695]: you have to keep these lines, see https://github.com/conan-io/conan-center-index/blob/master/docs/faqs.md#why-are-cmake-findconfig-files-and-pkg-config-files-not-packaged

File: recipes/qt/6.x.x/conanfile.py [hunk 3]
```diff
@@ -691,19 +698,11 @@ def package(self):
...
         tools.remove_files_by_mask(self.package_folder, "*.pdb*")
         tools.remove_files_by_mask(self.package_folder, "ensure_pro_file.cmake")
         os.remove(os.path.join(self.package_folder, "bin", "qt-cmake-private-install.cmake"))
 
-        for m in os.listdir(os.path.join(self.package_folder, "lib", "cmake")):
-            module = os.path.join(self.package_folder, "lib", "cmake", m, "%sMacros.cmake" % m)
-            helper_modules = glob.glob(os.path.join(self.package_folder, "lib", "cmake", m, "QtPublic*Helpers.cmake"))
-            if not os.path.isfile(module) and not helper_modules:
-                tools.rmdir(os.path.join(self.package_folder, "lib", "cmake", m))
-
```
Comment (ericLemanissier) [inline recipes/qt/6.x.x/conanfile.py:706]: same as above

File: recipes/qt/6.x.x/conanfile.py (hunk 1 above)
Comment (madebr) [reply recipes/qt/6.x.x/conanfile.py:620]: You mean conan does not generate the required cmake modules for build requirements?
Is there missing something else?
Imho, this deserves to be added to conan.

Comment (ericLemanissier) [reply recipes/qt/6.x.x/conanfile.py:620]: the thing is that you usually want different options for the `build` qt (basically just the minimum) from the `host` qt.

**Pull request closed without merging**